pkg/admissionpolicy/validate.go

```diff
@@ -78,12 +80,28 @@ func Validate(
 	bindings := policyData.bindings
 	engineResponse := engineapi.NewEngineResponse(resource, engineapi.NewValidatingAdmissionPolicy(&policy), nil)
 
+	var restMapper meta.RESTMapper
+	if client != nil {
```

Comment:
I think we have similar Code on multiple places. Should we create a reusable function for it?

Reply:
Let me check.

Reply:
I see that this code is duplicated [here](https://github.com/kyverno/kyverno/blob/ba7f6b432fbe7f5e83ac7b44a17652ce3b25410b/cmd/cli/kubectl-kyverno/processor/policy_processor.go#L252-L262). For this particular code, we check the client, but in the policy processor, we also check the cluster `p.Cluster`.
https://github.com/kyverno/kyverno/blob/ba7f6b432fbe7f5e83ac7b44a17652ce3b25410b/cmd/cli/kubectl-kyverno/processor/policy_processor.go#L242

It is important for the `test` command to work because it uses a fake client, so we added an extra condition to check if it is really connected to the cluster.